tooling/minimize/src/bb.rs

```diff
@@ -464,8 +465,26 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
                 })
                 .collect();
 
+            // Distinguish direct function calls or dynamic dispatch on a trait object.
+            let callee = if let rs::InstanceKind::Virtual(_trait, method) = instance.def {
+                // FIXME: This does not implement all receivers as allowed by `std::ops::DispatchFromDyn`.
+                // It doesn't (and also can't) properly adjust the type, however, MiniRust only requires the argument
```

Comment:
Why can't it do that?

Reply:
The proper Type for `&self` is a pointer with the size and alignement of 8. MiniRust can't know this at compile time.

Reply:
This doesn't quite make sense to me. Where does the 8 come from?

The Rust codegen backend also does this all at compile time so clearly it is possible